test/functional/apps/management/_scripted_fields.js

```diff
@@ -198,35 +198,44 @@ export default function ({ getService, getPageObjects }) {
       });
 
       it('should visualize scripted field in vertical bar chart', async function () {
-        const expectedChartValues = [
-          ['14', '31'],
-          ['10', '29'],
-          ['7', '24'],
-          ['11', '24'],
-          ['12', '23'],
-          ['20', '23'],
-          ['19', '21'],
-          ['6', '20'],
-          ['17', '20'],
-          ['30', '20'],
-          ['13', '19'],
-          ['18', '18'],
-          ['16', '17'],
-          ['5', '16'],
-          ['8', '16'],
-          ['15', '14'],
-          ['3', '13'],
-          ['2', '12'],
-          ['9', '10'],
-          ['4', '9'],
-        ];
         await filterBar.removeAllFilters();
         await PageObjects.discover.clickFieldListItemVisualize(scriptedPainlessFieldName);
         await PageObjects.header.waitUntilLoadingHasFinished();
 
-        await inspector.open();
-        await inspector.setTablePageSize(50);
-        await inspector.expectTableData(expectedChartValues);
+        if (await PageObjects.common.isOss()) {
+          // OSS renders a vertical bar chart and we check the data in the Inspect panel
+          const expectedChartValues = [
+            ['14', '31'],
+            ['10', '29'],
+            ['7', '24'],
+            ['11', '24'],
+            ['12', '23'],
+            ['20', '23'],
+            ['19', '21'],
+            ['6', '20'],
+            ['17', '20'],
+            ['30', '20'],
+            ['13', '19'],
+            ['18', '18'],
+            ['16', '17'],
+            ['5', '16'],
+            ['8', '16'],
+            ['15', '14'],
+            ['3', '13'],
+            ['2', '12'],
+            ['9', '10'],
+            ['4', '9'],
+          ];
+
+          await inspector.open();
+          await inspector.setTablePageSize(50);
+          await inspector.expectTableData(expectedChartValues);
+        } else {
+          // verify Lens opens a visualization
+          expect(await testSubjects.getVisibleTextAll('lns-dimensionTrigger')).to.contain(
+            'Average of ram_Pain1'
```

Comment:
I'm a bit confused to see `testSubjects.getVisibleTextAll('lns-dimensionTrigger')` + contains check:

<img width="1453" alt="Lens - Elastic 2020-11-05 17-19-08" src="https://user-images.githubusercontent.com/10977896/98267547-cd499980-1f8b-11eb-8b10-84eff54cc456.png">

So we get an array of 2 and check it contains `Average of ram_Pain1`, are we ok to ignore `@timestamp`?

Reply:
Yes, I thought about validating both @timestamp and the other field, but since this test is all about scripted fields I only verified the aggregated scripted field name.